demos/browser/app/meeting/meeting.ts

```diff
@@ -474,6 +475,32 @@ export class DemoMeetingApp implements AudioVideoObserver, DeviceChangeObserver
     });
   }
 
+  getNearestMediaRegion(): void {
+    const defaultNearestMediaRegion = "us-east-1";
+    new AsyncScheduler().start(
+      async (): Promise<void> => {
+        this.showProgress('progress-authenticate');
+        try {
+          const nearestMediaRegionResponse = await fetch(
+            `https://nearest-media-region.l.chime.aws`,
+            {
+              method: 'GET',
+            }
+          );
+          const nearestMediaRegionJSON = await nearestMediaRegionResponse.json();
+          const nearestMediaRegion = nearestMediaRegionJSON.region;
+          if(nearestMediaRegion !== null || nearestMediaRegion !== '') {
+            (document.getElementById('inputRegion') as HTMLInputElement).value = nearestMediaRegion;
+          }
+          else {
+            (document.getElementById('inputRegion') as HTMLInputElement).value = defaultNearestMediaRegion;
+          }
+        } catch (error) {
+          (document.getElementById('inputRegion') as HTMLInputElement).value = defaultNearestMediaRegion;
```

Comment:
It feels a bit odd to inform the client that `us-east-1` is the closest region and silently swallow an error. Could the client set a `defaultNearestMediaRegion` on setup? Could you at least log an error/warning here, and on line 496, that informs the client a issue occurred and a default will be returned?

Reply:
Thank you @mikeantonelli for your comment.
The getNearestMediaRegion() will GET the nearest media region as a fetch call to https://nearest-media-region.l.chime.aws and set the default option in the inputRegion drop down menu to the region being returned by this call. If the call to the API fails by any chance, the "us-east-1" will be selected as the default option as the closest region in the dropdown. Till this point, we are just 'selecting' the nearest media region. Meetings are not created/joined until the join button is clicked. So the inputRegion option can be changed by the client manually before clicking the join button to create/join the meeting.

Reply:
Thanks for the clarification

Reply:
We already set default region to use-east-1 so this is unnecessary. Also we should log error if the fetch failed.